packages/seacas/libraries/ioss/src/cgns/Iocgns_DatabaseIO.C

```diff
@@ -80,6 +80,28 @@ namespace {
     return has_decomp_flag;
   }
 
+  bool has_decomp_name_kluge(int cgns_file_ptr, int base, int zone, int zgc_idx)
+  {
+    // Check the `zgc_idx`-th ZGC node to see if the name matches the
+    // form described in the `name_is_decomp` function below.  We want to
+    // see if there are *any* names that match this form and if so, we can
+    // use the kluge; otherwise we can't and need to rely on other hueristics.
+    char                    connectname[CGNS_MAX_NAME_LENGTH + 1];
+    char                    donorname[CGNS_MAX_NAME_LENGTH + 1];
```

Comment:
Is there no way to use std::string so that we can avoid possible buffer over-runs?  is there strong checking for these max sizes to make sure that never happens?

Reply:
No, these get passed in to a C-API TPL which we don't have control over.  There is good checking in the TPL to avoid buffer over-runs.